### ISSUE TITLE
Invalid characters allowed when at the end of anyCharcter section.

### ISSUE THREAD

**Issue by Sharkaboi** [caseyryan/flutter_multi_formatter#38]
Hey, I was using this library and found a weird bug.
This occurs when invalid input is done at the end of the anyCharacter section. On next character, the invalid character is removed although the new valid character is lost.

To recreate:
- Use case : 
```
MaskedInputFormatter("#### #### #### ####", anyCharMatcher: RegExp(r'[0-9]+'))
```
- Enter `1234,56` where `,` is invalid. It shows as `1234 ,` then when 5 is typed, it is lost as it's replacing the invalid character. So output becomes `1234 6` after the whole input.

Also occurs with `CreditCardExpirationDateFormatter()`.  
Adding another formatter before to restrict characters to digits only causes different bug where output doesn't follow the mask.

**Comment by caseyryan** [comment]
do you use the null-safe version?

**Comment by Sharkaboi** [comment]
> 
> 
> do you use the null-safe version?

Library version : 2.0.1  
Flutter version : 2.0.3

**Comment by caseyryan** [comment]
Thanks, I'll have a look into as soon as I have time

**Comment by caseyryan** [comment]
Fixed in version 2.1.0
